streams/src/test/java/org/apache/kafka/streams/processor/internals/MockStreamsMetrics.java

```diff
@@ -0,0 +1,109 @@
+
+    Sensor commitTimeSensor;
+    Sensor pollTimeSensor;
+    Sensor processTimeSensor;
+    Sensor punctuateTimeSensor;
+    Sensor taskCreationSensor;
+    Sensor taskDestructionSensor;
+
+    public MockStreamsMetrics(Metrics metrics) {
+        this.metrics = metrics;
+        this.metricGrpName = "stream-metrics";
+        this.metricTags = new LinkedHashMap<>();
+
+        this.commitTimeSensor = metrics.sensor("commit-time");
+        this.pollTimeSensor = metrics.sensor("poll-time");
+        this.processTimeSensor = metrics.sensor("process-time");
+        this.punctuateTimeSensor = metrics.sensor("punctuate-time");
+        this.taskCreationSensor = metrics.sensor("task-creation");
+        this.taskDestructionSensor = metrics.sensor("task-destruction");
+
+    }
+
+    @Override
+    public Sensor addLatencySensor(String scopeName, String entityName, String operationName, Sensor.RecordLevel recordLevel, String... tags) {
+        Map<String, String> tagMap = new HashMap<>();
+        String metricGroupName = "stream-" + scopeName + "-metrics";
+
+        // first add the global operation metrics if not yet, with the global tags only
+        Sensor parent = metrics.sensor("sensorNamePrefix" + "." + scopeName + "-" + operationName, recordLevel);
+        addLatencyMetrics(metricGroupName, parent, "all", operationName, tagMap);
+
+        // add the store operation metrics with additional tags
+        Sensor sensor = metrics.sensor("sensorNamePrefix" + "." + scopeName + "-" + entityName + "-" + operationName, recordLevel, parent);
+        addLatencyMetrics(metricGroupName, sensor, entityName, operationName, tagMap);
+
+        return sensor;
+    }
+    private void addLatencyMetrics(String metricGrpName, Sensor sensor, String entityName, String opName, Map<String, String> tags) {
+        maybeAddMetric(sensor, metrics.metricName(entityName + "-" + opName + "-avg-latency-ms", metricGrpName,
+            "The average latency in milliseconds of " + entityName + " " + opName + " operation.", tags), new Avg());
+        maybeAddMetric(sensor, metrics.metricName(entityName + "-" + opName + "-max-latency-ms", metricGrpName,
+            "The max latency in milliseconds of " + entityName + " " + opName + " operation.", tags), new Max());
+        maybeAddMetric(sensor, metrics.metricName(entityName + "-" + opName + "-qps", metricGrpName,
+            "The average number of occurrence of " + entityName + " " + opName + " operation per second.", tags), new Rate(new Count()));
+    }
+
+    private void maybeAddMetric(Sensor sensor, MetricName name, MeasurableStat stat) {
+        if (!metrics.metrics().containsKey(name))
+            sensor.add(name, stat);
+    }
+
+
+    @Override
+    public void recordLatency(Sensor sensor, long startNs, long endNs) {
+    }
+
+    @Override
+    public Sensor sensor(String scopeName, String entityName, String operationName, Sensor.RecordLevel recordLevel, String... tags) {
```

Comment:
Tags are not used at all, actually do they need to be added? See my previous comment.